app/ts/components/Demo.tsx

```diff
@@ -0,0 +1,139 @@
+import { useComputed } from '@preact/signals'
+import { useAccount } from '../store/account.js'
+import { useAccountTokens } from '../store/tokens.js'
+import { useNetwork } from '../store/network.js'
+import { QueryToken } from './QueryToken.js'
+import { assertUnreachable } from '../library/utilities.js'
+
+export const DemoPage = () => {
+	return (
+		<div class='p-4'>
+			<div class='font-bold mt-4'>ActiveAddress</div>
+			<hr class='my-1' />
+			<Address />
+
+			<div class='font-bold mt-4'>Network ID</div>
+			<hr class='my-1' />
+			<Network />
+
+			<div class='font-bold mt-4'>Account Tokens</div>
+			<hr class='my-1' />
+			<AccountTokens />
+
+			<div class='font-bold mt-4'>Add Token</div>
+			<hr class='my-1' />
+			<AddTokenToAccount />
+		</div>
+	)
+}
+
+const Address = () => {
+	const { address, connect } = useAccount()
+
+	switch (address.value.state) {
+		case 'inactive':
+			return <Button text='Connect' onClick={() => connect()} />
+		case 'pending':
+			return <>connecting...</>
+		case 'rejected':
+			return <>error</>
```

Comment:
I currently just see `error` after clicking connect.  No details provided in the log or on the page.  Perhaps here we could just do:
```suggestion
		case 'rejected':
			return <>error: ${address.value.error.message}</>
```

Reply:
looks like `window.ethereum` does not have a `addListener` function if you are using other than MetaMask to connect. I've changed it to `on` as I found that Brave also supports this.

FWIW: the function seem to vary between vendor and I doesn't seem to be on 1193 specs